lms/djangoapps/instructor/views/api.py

```diff
@@ -772,6 +772,49 @@ def extract_user_info(user):
 @ensure_csrf_cookie
 @cache_control(no_cache=True, no_store=True, must_revalidate=True)
 @require_level('staff')
+def get_problem_responses(request, course_id):
+    """
+    Initiate generation of a CSV file containing all student answers
+    to a given problem.
+
+    Responds with JSON
+        {"status": "... status message ..."}
+
+    if initiation is successful (or generation task is already running).
+
+    Responds with BadRequest if problem URL is faulty.
+    """
+    course_key = CourseKey.from_string(course_id)
+    problem_url = request.GET.get('problem_url', '')
+
+    if problem_url[-4:] == ".xml":
+        problem_url = problem_url[:-4]
+
+    try:
+        course_key.make_usage_key_from_deprecated_string(problem_url)
+    except InvalidKeyError:
+        problem_not_found_status = _("Could not find problem with this URL.")
+        return JsonResponseBadRequest(problem_not_found_status)
```

Comment:
@bradenmacdonald I'm afraid that doesn't work (at least not for `i4x` URLs):

    ('COURSE_KEY: ', CourseLocator(u'OpenCraft', u'ID101', u'2015', None, None))
    ('PROBLEM_KEY.COURSE_KEY: ', CourseLocator(u'OpenCraft', u'ID101', None, None, None))

For some reason, `UsageKey.from_string(problem_url)` doesn't set `run` properly, so the equality check fails. This is also evident when I try to use `problem_key.course_key` with a format string:

> Traceback (most recent call last):
...
  File "instructor/views/api.py", line 796, in `get_problem_responses`
    `print('PROBLEM_KEY.COURSE_KEY: {}'.format(problem_key.course_key))`
  File "opaque_keys/edx/locator.py", line 52, in `__str__`
    `return unicode(self).encode('utf-8')`
  File "opaque_keys/__init__.py", line 167, in `__unicode__`
    `return self._to_deprecated_string()`
  File "opaque_keys/edx/locator.py", line 354, in `_to_deprecated_string`
    `return u'/'.join([self.org, self.course, self.run])`
**TypeError: sequence item 2: expected string or Unicode, NoneType found**

Maybe I could default to using `UsageKey.from_string` and conditionally fall back on one of the deprecated methods?